apps/os/backend/agent/slack-agent.ts

```diff
@@ -358,36 +364,69 @@ export class SlackAgent extends IterateAgent<SlackAgentSlices> implements ToolsI
       return [];
     }
 
-    const userMapping = await this.db.query.providerUserMapping.findFirst({
-      where: and(
-        eq(providerUserMapping.providerId, "slack-bot"),
-        eq(providerUserMapping.externalId, slackUserId),
-      ),
-      with: {
-        internalUser: true,
-      },
-    });
-
-    if (!userMapping?.internalUser) {
+    const estateId = this.databaseRecord.estateId;
+
+    const result = await this.db
+      .select({
+        userId: user.id,
+        userEmail: user.email,
+        userName: user.name,
+        slackUserId: providerUserMapping.externalId,
+        providerMetadata: providerUserMapping.providerMetadata,
+      })
+      .from(providerUserMapping)
+      .innerJoin(user, eq(providerUserMapping.internalUserId, user.id))
+      .innerJoin(organizationUserMembership, eq(user.id, organizationUserMembership.userId))
+      .innerJoin(organization, eq(organizationUserMembership.organizationId, organization.id))
+      .innerJoin(estate, eq(organization.id, estate.organizationId))
+      .where(
+        and(
+          eq(providerUserMapping.providerId, "slack-bot"),
+          eq(providerUserMapping.externalId, slackUserId),
+          eq(estate.id, estateId),
+        ),
+      )
+      .limit(1);
+
+    // If no result, user either doesn't exist or doesn't have access to this estate
+    if (!result[0]) {
+      waitUntil(
+        this.slackAPI.chat.postMessage({
+          channel: slackChannelId,
+          thread_ts: slackThreadId,
+          text: `Messaging iterate is not available for you now.`,
```

Comment:
Can we just remove this message? I think the most important thing is that the MCP tool is not called by guests, right? I don't think this here is a good user experience

See this slack message for more context https://iterate-com.slack.com/archives/C08R1SMTZGD/p1759329022030529?thread_ts=1759328429.403659&cid=C08R1SMTZGD